firebase-firestore/src/test/java/com/google/firebase/firestore/local/RemoteDocumentCacheTestCase.java

```diff
@@ -134,7 +136,7 @@ public void testSetAndReadLotsOfDocuments() {
   public void testSetAndReadDeletedDocument() {
     String path = "a/b";
     NoDocument deletedDoc = deletedDoc(path, 42);
-    add(deletedDoc);
+    add(deletedDoc, version(42));
```

Comment:
All of these tests use identical versions for the update_time and read_time which means we could have gotten these reversed and won't see it. Could you add a test that verifies read time doesn't get crossed with update time?

Reply:
I think it will become easier to add this test when https://github.com/firebase/firebase-android-sdk/pull/617 lands (which should be the next PR). I added a TODO for now and will resolve it when we gain the ability to exclude documents by read time.